internal/tiger/common/spinner.go

```diff
@@ -21,16 +21,30 @@ type Spinner interface {
 }
 
 // NewSpinner creates and returns a new [Spinner] for displaying animated
-// status messages. If output is a terminal, it uses bubbletea to dynamically
-// update the spinner and message in place. If output is not a terminal, it
-// prints each message on a new line without animation.
+// status messages. If the output is nil or [io.Discard], it returns a no-op
+// spinner. If output is a terminal, it uses bubbletea to dynamically update
+// the spinner and message in place. If output is not a terminal, it prints
+// each message on a new line without animation.
 func NewSpinner(output io.Writer, message string) Spinner {
+	if output == nil || output == io.Discard {
+		return newNopSpinner()
+	}
```

Comment:
This change makes it possible to use `WaitForService` in the MCP tools without worrying about the spinner (the MCP tools pass in a `nil` output parameter, which results in a no-op spinner).